File: TrackwiseAPI/Models/Entities/Delivery.cs
using Microsoft.AspNetCore.Routing.Constraints;
using Microsoft.EntityFrameworkCore.ValueGeneration.Internal;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TrackwiseAPI.Models.Entities
{
    public class Delivery
    {
        [Key]
        public string Delivery_ID { get; set; }
        public double Delivery_Weight { get; set; }
        public double? Initial_Mileage { get; set; }
        public double? Final_Mileage { get; set; }
        public double? TotalFuel { get; set; }
        public bool WeightCaptured { get; set; }
        public bool MileageCaptured { get; set; }

        //Foreign Key for Job
        public string  Job_ID { get; set; }
        [ForeignKey("Job_ID")]
        public Job Job { get; set; }

        //Foreign Key for Driver
        public string Driver_ID { get; set; }
        [ForeignKey("Driver_ID")]
        public Driver Driver { get; set; }

        //Foreign Key for Truck
        public string TruckID { get; set; }
        [ForeignKey("TruckID")]
        public Truck Truck { get; set; }

        //Foreign Key for Trailer
        public string TrailerID { get; set; }
        [ForeignKey("TrailerID")]
        public Trailer Trailer { get; set; }
        public string Delivery_Status_ID { get; set; }
        [ForeignKey("Delivery_Status_ID")]
        public DeliveryStatus DeliveryStatus { get; set; }

        public ICollection<Document> Documents { get; set; }

    }
}

File: TrackwiseIonic/src/app/shared/Delivery.ts
export interface Job {
  job_ID: string;
  startDate: Date;
  dueDate: Date;
  pickupLocation: string;
  dropoffLocation: string;
  type: string;
  mapURL:string;
}

// Update the Delivery interface to include the Job properties
export interface Delivery {
  delivery_ID: string;
  delivery_Weight: number;
  delivery_Status_ID: string;
  DeliveryStatus: DeliveryStatus;
  JobStatus:JobStatus;
  job_Status_ID:string;
  driver_ID:string;
  Driver:Driver;
  truckID:string;
  Truck:Truck;
  trailerID:string;
  Trailer:Trailer;
  job_ID:string;
  jobs: Job; // Use the Job interface here
}

export interface DeliveryStatus {
  delivery_Status_ID: string;
  name: string;
  description: string;
  Deliveries: Delivery[];
}
export interface JobStatus {
  job_Status_ID: string;
  name: string;
  description: string;
  Deliveries: Delivery[];
}

export interface Driver {
  driver_ID: string;
  name: string;
  Deliveries: Delivery[];
}
export interface Trailer {
  trailerID: string;
  name: string;
  Deliveries: Delivery[];
}
export interface Truck {
  truckID: string;
  name: string;
  Deliveries: Delivery[];
}
